src/main/java/com/echobox/api/linkedin/connection/versioned/VersionedVideoConnection.java

```diff
@@ -34,13 +57,96 @@ public class VersionedVideoConnection extends VersionedConnection {
   private static final String VIDEOS = "/videos";
   private static final String ACTION_KEY = "action";
   private static final String INITIALIZE_UPLOAD = "initializeUpload";
+  private static final String FINALIZE_UPLOAD = "finalizeUpload";
+  
+  /**
+   * Headers
+   */
+  private static final String HEADER_ETAG = "etag";
   
   /**
    * Instantiates a new connection base.
    *
    * @param linkedinClient the LinkedIn client
    */
-  protected VersionedVideoConnection(VersionedLinkedInClient linkedinClient) {
+  public VersionedVideoConnection(VersionedLinkedInClient linkedinClient) {
     super(linkedinClient);
   }
+  
+  public URN uploadVideo(InitializeUploadRequest initializeUploadRequest, String filePath)
+      throws IOException {
+  
+    Path videoFilePath = Paths.get(filePath);
+    long fileSizeBytes = Files.size(videoFilePath);
+    ValidationUtils.validateVideoFile(fileSizeBytes);
+  
+    initializeUploadRequest.getInitializeUploadRequest().setFileSizeBytes(fileSizeBytes);
+    InitializeUploadResponse initializeUploadResponse = initializeUpload(initializeUploadRequest);
+    InitializeUploadResponse.Value value = initializeUploadResponse.getValue();
+    
+    List<String> uploadedPartIds = uploadVideoFile(filePath, value.getUploadInstructions());
+    
+    FinalizeUploadRequest finalizeUploadRequest =
+        new FinalizeUploadRequest(value.getVideo(), value.getUploadToken(), uploadedPartIds);
+    finalizeUpload(finalizeUploadRequest);
+    
+    return value.getVideo();
+  }
+  
+  public InitializeUploadResponse initializeUpload(
+      InitializeUploadRequest initializeUploadRequest) {
+    return linkedinClient.publish(VIDEOS, InitializeUploadResponse.class, initializeUploadRequest,
+        Parameter.with(ACTION_KEY, INITIALIZE_UPLOAD));
+  }
+  
+  public List<String> uploadVideoFile(String filePath,
+      List<InitializeUploadResponse.UploadInstruction> uploadInstructions) throws IOException {
+    
+    File file = new File(filePath);
+    byte[] fileBytes = convertToBytes(file);
+    
+    List<String> uploadPartIds = new ArrayList<>();
+    for (InitializeUploadResponse.UploadInstruction instruction : uploadInstructions) {
+      String etag = uploadVideoFileChunk(filePath, fileBytes, instruction);
+      uploadPartIds.add(etag);
+    }
+    
+    return uploadPartIds;
+  }
+  
+  public String uploadVideoFileChunk(String filePath, byte[] fileBytes,
+      InitializeUploadResponse.UploadInstruction instruction) throws IOException {
+    WebRequestor webRequestor = linkedinClient.getWebRequestor();
+    
+    Map<String, String> requestHeaders = new HashMap<>();
+    requestHeaders.put(DefaultVersionedLinkedInClient.HEADER_NAME_VERSION,
+        DefaultVersionedLinkedInClient.DEFAULT_VERSIONED_MONTH);
```

Comment:
I think we need to get the `versionedMonth `from underlying `linkedInClient`.

Reply:
Valid point. We currently don't have a method to retrieve the `versionedMonth` and by implementing it means we have to make changes to the `DefaultLinkedInClient`, which is not ideal. I will throw a `UnsupportedOperationException` from that method considering we're deprecating that client.